src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,951 @@
+            Document originalDocument, Document document, Dictionary<TIdentifierNameSyntax, IParameterSymbol> mappingDictionary,
+            IMethodSymbol methodSymbol, SyntaxNode containingMethod, TExpressionSyntax expression,
+            bool allOccurrences, string parameterName,
+            List<SyntaxNode> invocations, CancellationToken cancellationToken)
+        {
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var semanticFacts = document.GetRequiredLanguageService<ISemanticFactsService>();
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var editor = new SyntaxEditor(root, generator);
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var insertionIndex = GetInsertionIndex(compilation, methodSymbol, syntaxFacts, containingMethod);
+            var invocationSemanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            foreach (var invocation in invocations)
+            {
+                var expressionEditor = new SyntaxEditor(expression, generator);
+
+                SeparatedSyntaxList<SyntaxNode> invocationArguments;
+                if (methodSymbol.MethodKind == MethodKind.Constructor)
+                {
+                    invocationArguments = syntaxFacts.GetArgumentsOfObjectCreationExpression(invocation);
+                }
+                else
+                {
+                    invocationArguments = syntaxFacts.GetArgumentsOfInvocationExpression(invocation);
+                }
+
+                var parameterToArgumentMap = MapParameterToArgumentsAtInvocation(semanticFacts, invocationArguments, invocationSemanticModel, cancellationToken);
+
+                SyntaxNode argumentListSyntax;
+                if (methodSymbol.MethodKind is MethodKind.Constructor)
+                {
+                    argumentListSyntax = syntaxFacts.GetArgumentListOfObjectCreationExpression(invocation);
+                }
+                else
+                {
+                    argumentListSyntax = syntaxFacts.GetArgumentListOfInvocationExpression(invocation);
+                }
+
+                editor.ReplaceNode(argumentListSyntax, (currentArgumentListSyntax, _) =>
+                {
+                    var updatedInvocationArguments = syntaxFacts.GetArgumentsOfArgumentList(currentArgumentListSyntax);
+                    var updatedExpression = CreateNewArgumentExpression(expressionEditor,
+                        syntaxFacts, mappingDictionary, parameterToArgumentMap, updatedInvocationArguments);
+                    var allArguments = AddArgumentToArgumentList(updatedInvocationArguments, generator,
+                        updatedExpression.WithAdditionalAnnotations(Formatter.Annotation), insertionIndex);
+                    return syntaxFacts.UpdateArgumentListSyntax(currentArgumentListSyntax, allArguments);
```

Comment:
i feel lke you probably added this syntax fact.  can you make this into an abstract method isntead.  this doesn't really fit in.

Reply:
removed